### ISSUE TITLE
Local discovery between two phones with Hotspot of one not working, global discovery very slow

### ISSUE THREAD

**Issue by martin-braun** [Catfriend1/syncthing-android#905]
### Description of the issue

I have a setup of 3 devices that have Syncthing installed: 

- Pixel 3a (provides a local network with a Hotspot)
- Galaxy S10
- MacBook

I'm using your fork on the mobile devices. The MacBook and the Galaxy S10 is connected to the Hotspot of the Pixel 3a. The Syncthing-Fork on Pixel 3a and Galaxy S10 is currently in force mode, so no run conditions play a role here. I setup a folder to be synced on all devices.

In the device tab, they will only connect with the MacBook. If I disable Syncthing on the MacBook, it will not solve the issue.

All devices have NAT Traversal, Global Discovery and Relaying off, as I want to transmit only via local network. 

If I enable global discovery on the mobile devices, they will connect with each other after a very long discovery time of almost 5 minutes, eventually.

### Reproduction Steps

- Two Android Devices with a shared folder
- Both devices run Syncthing-Fork with NAT Traversal / Global Discovery and Enable Relaying being off (only Local Discovery is on)
- One device has a hotspot open that the other device is connected to
- Both devices are added and share a folder (everything is setup on each device properly)
- Force mode is enabled
- Wait 10 minutes to see nothing is happening
- Enable Global Discovery (for test purposes)
- Wait 5 minutes to see they connect eventually

### Version Information
- App Version: 1.20.3.1
- Syncthing Version: v1.20.3

#### Pixel 3a
- Android Version: Android 12 (Kernel 4.9.292-gab4493f31457)
- Device manufacturer: Google
- Device model: Pixel 3a
- Device platform info (`adb shell "getprop | grep ro.product"`):
```sh
[ro.product.ab_ota_partitions]: [vendor,boot,system,vbmeta,dtbo,product,system_ext,abl,aop,cmnlib,cmnlib64,devcfg,hyp,keymaster,qupfw,tz,xbl,xbl_config,modem]
[ro.product.board]: [sargo]
[ro.product.bootimage.brand]: [google]
[ro.product.bootimage.device]: [sargo]
[ro.product.bootimage.manufacturer]: [Google]
[ro.product.bootimage.model]: [Pixel 3a]
[ro.product.bootimage.name]: [sargo]
[ro.product.brand]: [google]
[ro.product.build.date]: [Mon May  9 18:24:52 UTC 2022]
[ro.product.build.date.utc]: [1652120692]
[ro.product.build.fingerprint]: [google/sargo/sargo:12/SP2A.220505.006/8561491:user/release-keys]
[ro.product.build.id]: [SP2A.220505.006]
[ro.product.build.tags]: [release-keys]
[ro.product.build.type]: [user]
[ro.product.build.version.incremental]: [8561491]
[ro.product.build.version.release]: [12]
[ro.product.build.version.release_or_codename]: [12]
[ro.product.build.version.sdk]: [32]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.device]: [sargo]
[ro.product.first_api_level]: [28]
[ro.product.locale]: [en-US]
[ro.product.manufacturer]: [Google]
[ro.product.model]: [Pixel 3a]
[ro.product.name]: [sargo]
[ro.product.odm.brand]: [google]
[ro.product.odm.device]: [sargo]
[ro.product.odm.manufacturer]: [Google]
[ro.product.odm.model]: [Pixel 3a]
[ro.product.odm.name]: [sargo]
[ro.product.product.brand]: [google]
[ro.product.product.device]: [sargo]
[ro.product.product.manufacturer]: [Google]
[ro.product.product.model]: [Pixel 3a]
[ro.product.product.name]: [sargo]
[ro.product.system.brand]: [google]
[ro.product.system.device]: [generic]
[ro.product.system.manufacturer]: [Google]
[ro.product.system.model]: [mainline]
[ro.product.system.name]: [mainline]
[ro.product.system_ext.brand]: [google]
[ro.product.system_ext.device]: [sargo]
[ro.product.system_ext.manufacturer]: [Google]
[ro.product.system_ext.model]: [Pixel 3a]
[ro.product.system_ext.name]: [sargo]
[ro.product.vendor.brand]: [google]
[ro.product.vendor.device]: [sargo]
[ro.product.vendor.manufacturer]: [Google]
[ro.product.vendor.model]: [Pixel 3a]
[ro.product.vendor.name]: [sargo]
[ro.product.vendor_dlkm.brand]: [google]
[ro.product.vendor_dlkm.device]: [sargo]
[ro.product.vendor_dlkm.manufacturer]: [Google]
[ro.product.vendor_dlkm.model]: [Pixel 3a]
[ro.product.vendor_dlkm.name]: [sargo]
```

#### Galaxy S10
- Android Version: Android 12 (Kernel 4.14.113-24166302)
- Device manufacturer: Samsung
- Device model: Galaxy S10
- Device platform info (`adb shell "getprop | grep ro.product"`):
```sh
[ro.product.board]: [exynos9820]
[ro.product.bootimage.brand]: [samsung]
[ro.product.bootimage.device]: [beyond1]
[ro.product.bootimage.manufacturer]: [samsung]
[ro.product.bootimage.model]: [SM-G973F]
[ro.product.bootimage.name]: [beyond1ltexx]
[ro.product.brand]: [samsung]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.device]: [beyond1]
[ro.product.first_api_level]: [28]
[ro.product.locale]: [en-GB]
[ro.product.manufacturer]: [samsung]
[ro.product.model]: [SM-G973F]
[ro.product.name]: [beyond1lteeea]
[ro.product.odm.brand]: [samsung]
[ro.product.odm.device]: [beyond1]
[ro.product.odm.manufacturer]: [samsung]
[ro.product.odm.model]: [SM-G973F]
[ro.product.odm.name]: [beyond1ltexx]
[ro.product.system.brand]: [samsung]
[ro.product.system.device]: [beyond1]
[ro.product.system.manufacturer]: [samsung]
[ro.product.system.model]: [SM-G973F]
[ro.product.system.name]: [beyond1ltexx]
[ro.product.system_ext.brand]: [samsung]
[ro.product.system_ext.device]: [beyond1]
[ro.product.system_ext.manufacturer]: [samsung]
[ro.product.system_ext.model]: [SM-G973F]
[ro.product.system_ext.name]: [beyond1ltexx]
[ro.product.vendor.brand]: [samsung]
[ro.product.vendor.device]: [beyond1]
[ro.product.vendor.manufacturer]: [samsung]
[ro.product.vendor.model]: [SM-G973F]
[ro.product.vendor.name]: [beyond1ltexx]
[ro.product.vendor_dlkm.brand]: [samsung]
[ro.product.vendor_dlkm.device]: [beyond1]
[ro.product.vendor_dlkm.manufacturer]: [samsung]
[ro.product.vendor_dlkm.model]: [SM-G973F]
[ro.product.vendor_dlkm.name]: [beyond1ltexx]
[ro.product_ship]: [true]
```

### Android Log (Pixel 3a)

Syncthing-Fork running and force enabled on Galaxy S10 right before starting Syncthing-Fork and enabling force on Pixel 3a to sync two folders:
https://gist.githubusercontent.com/martin-braun/419be2140f1773d42fd9f971ed98722f/raw/d33b7cb2b3a22ce56e1736280c218c1f9d41fb12/logcat.log

**Comment by Catfriend1** [comment]
Hi,

this is duplicate of issue #801 . Local discovery is limited by Android. You need to configure static IP addresses.

**Comment by martin-braun** [comment]
@Catfriend1 Thanks for looking into this. Seems familiar indeed. Could you please open the referenced issue at least, even when you cannot work on this? 

What I also don't understand is that our phones were connected, but now they never are again. So we can only sync when my MacBook is on as well. Not optimal.
